File: frontend/src/backend/FoodPurchaseService.js
import {config, appFetch} from './appFetch';

export const createFoodPurchase = (foodPurchase, onSuccess, onErrors) => {
    appFetch('/foodPurchase/createFoodPurchase', config('POST', foodPurchase),
        onSuccess, onErrors);
}

export const updateFoodPurchase = (foodPurchaseId, foodPurchase, onSuccess, onErrors) => {
    appFetch(`/foodPurchase/${foodPurchaseId}`, config('PUT', foodPurchase),
        onSuccess, onErrors);
}

export const getFoodPurchaseById = (foodPurchaseId, onSuccess, onErrors) => {
    appFetch(`/foodPurchase/${foodPurchaseId}`, config('GET'), onSuccess, onErrors);
}

export const deleteFoodPurchase = (foodPurchaseId, onSuccess, onErrors) => {
    appFetch(`/foodPurchase/${foodPurchaseId}/deleteFoodPurchase`, config('DELETE'), onSuccess, onErrors);
}

export const getAllFoodPurchases = ({page, productName, supplier, startDate, endDate}, onSuccess, onErrors) => {
    let path = `/foodPurchase/allFoodPurchases?page=${page}&size=5`;

    path += productName?.length > 0 ? `&productName=${encodeURIComponent(productName)}` : "";
    path += supplier?.length > 0 ? `&supplier=${encodeURIComponent(supplier)}` : "";
    path += startDate ? `&startDate=${encodeURIComponent(startDate.toISOString().substring(0,10))}` : "";
    path += endDate ? `&endDate=${encodeURIComponent(endDate.toISOString().substring(0,10))}` : "";

    appFetch(path, config('GET'), onSuccess, onErrors);
}

export const getListOfAllFoodPurchases = (onSuccess, onErrors) => {
    appFetch('/foodPurchase/getListOfFoodPurchases', config('GET'), onSuccess, onErrors);
}

export const getAllAvailableFoodBatches = (onSuccess, onErrors) => {
    appFetch('/foodPurchase/getAvailableFoodBatches', config('GET'), onSuccess, onErrors);
}

File: backend/src/main/java/es/udc/paproject/backend/model/services/FoodPurchaseService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.FoodPurchase;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.PermissionException;
import es.udc.paproject.backend.rest.dtos.FoodPurchaseDTOs.FoodPurchaseDto;

import java.math.BigDecimal;
import java.util.List;

public interface FoodPurchaseService {

    void createFoodPurchase(FoodPurchase foodPurchase);

    FoodPurchase updateFoodPurchase(Long id, String productName, String ingredients, String supplier, Integer kilos,
                                    BigDecimal unitPrice) throws InstanceNotFoundException;

    void deleteFoodPurchase(Long userId, Long id) throws InstanceNotFoundException, PermissionException;

    FoodPurchase getFoodPurchaseById(Long userId, Long id) throws InstanceNotFoundException;

    Block<FoodPurchase> getAllFoodPurchases(Long userId, String productName, String supplier, String startDate, String endDate,
        int page, int size) throws InstanceNotFoundException;

    List<FoodPurchase> getListOfAllFoodPurchases(Long userId) throws InstanceNotFoundException;

    List<FoodPurchaseDto> getAllAvailablesFoodBatches(Long userId) throws InstanceNotFoundException;
}
